prow/plugins/project/project.go

```diff
@@ -0,0 +1,270 @@
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, person := range projectMaintainers {
+		login := github.NormLogin(e.User.Login)
+		if github.NormLogin(person.Login) == login {
+			found = true
+			break
+		}
+	}
+	if !found {
+		// not in the project maintainers team
+		msg := fmt.Sprintf(mustBeSigLead, org, project.ProjectMaintainersTeam, org, project.ProjectMaintainersTeam)
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	var projects []Project
+
+	repos, err := gc.GetRepos(org, false)
+	if err != nil {
+		log.WithError(err).Errorf("Error listing all repos in org %s", org)
+		return err
+	}
+	// Get all projects for all repos
+	for _, repo := range repos {
+		repoProjects, err := gc.GetRepoProjects(org, repo.Name)
+		if err != nil {
+			log.WithError(err).Errorf("Error listing all projects under repo %s", repo.Name)
+			return err
+		}
+		projects = append(projects, repoProjects)
+	}
+
+	// Get all projects for this org
+	orgProjects, err := gc.GetOrgProjects(org)
+	if err != nil {
+		log.WithError(err).Errorf("Error getting projects for org %s", org)
+		return err
+	}
+	projects = append(projects, orgProjects)
+
+	// Get all columns for each project
+	projectColumnMap map[string][]github.ProjectColumn
+	for _, project := range projects {
+		projectColumns, err := gs.GetProjectColumns(project.ID)
+		if err != nil {
+			log.WithError(err).Errorf("Error listing all columns under project %s", project.Name)
+			return err
+		}
+		projectColumnMap[project.Name] = projectColumns
+	}
+
+	projectMap := buildProjectMap(projects)
+	projectID, ok := projectMap[proposedProject]
+
+	if proposedProject == clearKeyword {
+		projectToClear := matches[2]
+		projectToClearID, ok := projectMap[projectToClear]
+		if ok {
+			existingProjectCard, err := gc.GetProjectCard(projectToClearID, e.Number)
+			if err != nil {
+				log.WithError(err).Errorf("Error geting existing project card under project ID %d for issue/PR %s", projectToClearID, e.Number)
+				return nil
+			}
+
+			if err := gc.DeleteProjectCard(existingProjectCard.ID); err != nil {
+				log.WithError(err).Errorf("Error removing project card %d from project %s for %s/%s#%d.", existingProjectCard.ID, project, org, repo, e.Number)
+			}
+			return nil
+		}
+	}
+
+	if !ok {
+		slice := make([]string, 0, len(projectMap))
+		for k := range projectMap {
+			slice = append(slice, fmt.Sprintf("`%s`", k))
+		}
+		sort.Strings(slice)
+
+		msg := fmt.Sprintf(invalidProject, strings.Join(slice, ", "))
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	columnFound := false
+	projectColumns := projectColumnMap[proposedProject]
+	for _, c := range projectColumns {
+		if c.ID == proposedColumn {
+			columnFound = true
+			break
+		}
+	}
+
+	// If proposedColumn is not found (or not provided), add to the "To do",
```

Comment:
Maybe we should expose the default column as a config option and do nothing if it is not set?